src/plugins/wizard/public/application/utils/get_top_nav_config.test.tsx

```diff
@@ -0,0 +1,155 @@
+
+  test('return undefined when savedWizardVis is null', async () => {
+    savedWizardVis = null;
+    const onSave = getOnSave(
+      savedWizardVis,
+      originatingApp,
+      visualizationIdFromUrl,
+      dispatch,
+      mockServices
+    );
+    const onSaveResult = await onSave(onSaveProps);
+
+    expect(onSaveResult).toBeUndefined();
+  });
+
+  test('savedWizardVis get saved correctly', async () => {
+    const onSave = getOnSave(
+      savedWizardVis,
+      originatingApp,
+      visualizationIdFromUrl,
+      dispatch,
+      mockServices
+    );
+    const onSaveReturn = await onSave(onSaveProps);
+    expect(savedWizardVis).toMatchInlineSnapshot(`
+      Object {
+        "copyOnSave": false,
+        "description": "new description",
+        "id": "1",
+        "save": [MockFunction] {
+          "calls": Array [
+            Array [
+              Object {
+                "confirmOverwrite": false,
+                "isTitleDuplicateConfirmed": false,
+                "onTitleDuplicate": [MockFunction],
+                "returnToOrigin": true,
+              },
+            ],
+          ],
+          "results": Array [
+            Object {
+              "type": "return",
+              "value": "1",
+            },
+          ],
+        },
+        "searchSourceFields": Object {},
+        "styleState": "",
+        "title": "new title",
+        "version": 0,
+        "visualizationState": "",
+      }
+    `);
+    expect(onSaveReturn?.id).toBe('1');
+  });
+
+  test('savedWizardVis does not change title with a null id', async () => {
+    savedWizardVis.save = jest.fn().mockReturnValue(null);
+    const onSave = getOnSave(
+      savedWizardVis,
+      originatingApp,
+      visualizationIdFromUrl,
+      dispatch,
+      mockServices
+    );
+    const onSaveResult = await onSave(onSaveProps);
+    expect(savedWizardVis.title).toBe('save wizard wiz title');
+    expect(onSaveResult?.id).toBeNull();
+  });
+
+  test('create a new wizard from dashboard', async () => {
+    savedWizardVis.id = null;
+    savedWizardVis.save = jest.fn().mockReturnValue('2');
+    originatingApp = 'dashboard';
+    onSaveProps.returnToOrigin = true;
+
+    const onSave = getOnSave(
+      savedWizardVis,
+      originatingApp,
+      visualizationIdFromUrl,
+      dispatch,
+      mockServices
+    );
+    const onSaveResult = await onSave(onSaveProps);
+    expect(onSaveResult?.id).toBe('2');
+    expect(dispatch).toBeCalledTimes(0);
+  });
+
+  test('edit an exising wizard from dashboard', async () => {
+    savedWizardVis.copyOnSave = false;
+    originatingApp = 'dashboard';
+    onSaveProps.returnToOrigin = true;
+    const onSave = getOnSave(
+      savedWizardVis,
+      originatingApp,
+      visualizationIdFromUrl,
+      dispatch,
+      mockServices
+    );
+    const onSaveResult = await onSave(onSaveProps);
+    expect(onSaveResult?.id).toBe('1');
+    expect(mockServices.application.navigateToApp).toBeCalledTimes(1);
```

Comment:
This test case is slightly fragile, in that we're just checking that the id didn't change. It could be more robust if one of the other properties actually updates, as you did in the `savedWizardVis get saved correctly` test case.

Reply:
Thanks, i modified the test by actually changing the description field and verifying if the description property gets properly updated.